### PR TITLE
LSR-886 Asset Display Screenlet: new features (Android)

### PR DESCRIPTION
**Pull request by dgarciasarai**

### CODE REVIEW

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java [hunk 1]
```diff
@@ -235,4 +235,5 @@ public void setScreenlet(BaseScreenlet screenlet) {
 	private ProgressBar progressBar;
 	private TextView progressText;
 	private TextView title;
+	private Matrix m;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java]: matrix :P

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/asset/display/AssetDisplayView.java [hunk 2]
```diff
@@ -64,6 +64,15 @@ public void setScreenlet(BaseScreenlet screenlet) {
 		this.screenlet = screenlet;
 	}
 
+	@Override
+	public void showFinishOperation(BaseScreenlet screenlet) {
+		container.removeAllViews();
+		container.addView(screenlet, LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT);
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/asset/display/AssetDisplayView.java]: this is hacky, why do you need it?

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java [hunk 3]
```diff
@@ -106,9 +113,11 @@ private void render() {
 	public void onClick(View v) {
 		if (v.getId() == R.id.liferay_previous_page) {
 			changeCurrentPage(-1);
-		} else if (v.getId() == R.id.liferay_next_page) {
+		}
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java]: watch out format

File: android/library/src/main/java/com/liferay/mobile/screens/blogs/BlogsEntryDisplayScreenlet.java [hunk 4]
```diff
@@ -19,6 +19,8 @@
 public class BlogsEntryDisplayScreenlet extends BaseScreenlet<BlogsEntryDisplayViewModel, AssetDisplayInteractorImpl>
 	implements AssetDisplayListener {
 
+	public static final String LOAD_ASSET_ACTION = "LOAD_ASSET_ACTION";
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/blogs/BlogsEntryDisplayScreenlet.java]: LOAD BLOG?

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayFactory.java [hunk 5]
```diff
@@ -36,6 +38,15 @@ public BaseScreenlet getScreenlet(Context context, AssetEntry assetEntry, HashMa
 					screenlet.render(layoutId);
 					return screenlet;
 				}
+			case "com.liferay.blogs.kernel.model.BlogsEntry":
+
+				BlogsEntryDisplayScreenlet blogsScreenlet = new BlogsEntryDisplayScreenlet(context);
+				Integer layoutId = layouts.get(blogsScreenlet.getClass().getName());
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayFactory.java]: uhm... why not use the typical way of getting the default layout?

**Comment by nhpatt** [comment]
Commented a few things :)

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/asset/display/AssetDisplayView.java (hunk 2 above)
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/asset/display/AssetDisplayView.java]: I think if there is an old screenlet in the container, we should remove it first and then add the new one.

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java (hunk 1 above)
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java]: 😆

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/asset/display/AssetDisplayView.java (hunk 2 above)
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/asset/display/AssetDisplayView.java]: But, why do you need to remove and add? Can't you update the screenlet?

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java (hunk 3 above)
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/dlfile/display/PdfDisplayView.java]: ok, sorry

File: android/library/src/main/java/com/liferay/mobile/screens/blogs/BlogsEntryDisplayScreenlet.java (hunk 4 above)
Comment (dgarciasarai) [inline android/library/src/main/java/com/liferay/mobile/screens/blogs/BlogsEntryDisplayScreenlet.java]: Yes! Sorry again

**Pull request merged**